src/main/java/com/autotune/analyzer/services/GenerateRecommendations.java

```diff
@@ -0,0 +1,362 @@
+                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.EXPERIMENT_NAME_MANDATORY);
+                return;
+            }
+
+            // Check if interval_end_time is provided
+            if (intervalEndTimeStr == null || intervalEndTimeStr.isEmpty()) {
+                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.INTERVAL_END_TIME_MANDATORY);
+                return;
+            }
+            if (!Utils.DateUtils.isAValidDate(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT, intervalEndTimeStr)) {
+                interval_end_time = null;
+                sendErrorResponse(
+                        response,
+                        new Exception(AnalyzerErrorConstants.APIErrors.ListRecommendationsAPI.INVALID_TIMESTAMP_EXCPTN),
+                        HttpServletResponse.SC_BAD_REQUEST,
+                        String.format(AnalyzerErrorConstants.APIErrors.ListRecommendationsAPI.INVALID_TIMESTAMP_MSG, intervalEndTimeStr)
+                );
+                return;
+            } else {
+                interval_end_time = Utils.DateUtils.getTimeStampFrom(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT, intervalEndTimeStr);
+                interval_start_time = RecommendationUtils.getMonitoringStartTime(
+                        interval_end_time,
+                        Double.valueOf(String.valueOf(KruizeConstants.RecommendationEngineConstants
+                                .DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS * 24)));
+                LOGGER.info("{} - {}", interval_start_time, interval_end_time);
+            }
+            LOGGER.debug("experiment_name : {} and interval_end_time : {} ", experiment_name, intervalEndTimeStr);
+
+            List<ExperimentResultData> experimentResultDataList = new ArrayList<>();
+            ExperimentResultData experimentResultData = null;
+            Map<String, KruizeObject> mainKruizeExperimentMAP = new ConcurrentHashMap<>();
+            KruizeObject kruizeObject = null;
+            DataSourceInfo dataSourceInfo = null;
+            try {
+                long interval_end_time_epoc = interval_end_time.getTime() / 1000 - (interval_end_time.getTimezoneOffset() * 60);
+                ;
+                long interval_start_time_epoc = interval_start_time.getTime() / 1000 - (interval_start_time.getTimezoneOffset() * 60);
+                ;
+
+                new ExperimentDBService().loadExperimentFromDBByName(mainKruizeExperimentMAP, experiment_name);
+                if (null != mainKruizeExperimentMAP.get(experiment_name)) {
+                    kruizeObject = mainKruizeExperimentMAP.get(experiment_name);
+                    dataSourceInfo = new ExperimentDBService().loadDataSourceFromDBByName(kruizeObject.getDataSource());
```

Comment:
@khansaad 
Could you please confirm whether the experiment data source is being retrieved from the database? It appears that I was receiving null values.

Reply:
Updated the converter method to include the datasource value. Let me know if it's working now